site/gatsby-site/cypress/integration/downloadIndex.js

```diff
@@ -0,0 +1,92 @@
+import { gql } from '@apollo/client';
+import path from 'path';
+
+describe('Download Algolia Index', () => {
+  const url = '/downloadIndex';
+
+  const downloadsFolder = Cypress.config('downloadsFolder');
+
+  it('Should return a properly constructed Algolia index', () => {
+    cy.visit(url);
+
+    cy.query({
+      query: gql`
+        query {
+          reports(limit: 9999) {
+            _id
+            cloudinary_id
+            date_downloaded
+            date_modified
+            date_published
+            date_submitted
+            description
+            epoch_date_downloaded
+            epoch_date_modified
+            epoch_date_published
+            epoch_date_submitted
+            epoch_incident_date
+            flag
+            image_url
+            incident_date
+            incident_id
+            language
+            ref_number
+            report_number
+            source_domain
+            text
+            title
+            url
+          }
+        }
+      `,
+    }).then(({ data: { reports } }) => {
+      cy.get('[data-cy=download]').click();
+
+      cy.readFile(path.join(downloadsFolder, 'index.json'), { timeout: 15000 }).then((index) => {
+        expect(index).to.have.length(reports.length);
+        expect(index.find((r) => r.report_number == 19)).to.deep.equal({
+          objectID: '5d34b8c29ced494f010ed46c',
```

Comment:
Maybe you need to get the whole index to ensure this record is in the set?

Reply:
Not sure I follow. This tests downloads the entire index JSON file and  checks that the report number 19 has proper structure and values.

Reply:
If you set the number of records to return to a small number, then this specific record might not be returned.